test/test_host.go

```diff
@@ -63,26 +63,27 @@ func (s *HostSuite) TestAddFailingJob(t *c.C) {
 	t.Assert(h.AddJob(job), c.IsNil)
 
 	// check we get a create then error event
-	actual := make([]*host.Event, 0, 2)
+	actual := make(map[string]*host.Event, 2)
 loop:
 	for {
 		select {
 		case e, ok := <-events:
 			if !ok {
 				t.Fatalf("job event stream closed unexpectedly: %s", stream.Err())
 			}
-			actual = append(actual, e)
+			if _, ok := actual[e.Event]; ok {
+				t.Fatalf("unexpected event: %v", e)
+			}
+			actual[e.Event] = e
 			if len(actual) >= 2 {
 				break loop
 			}
 		case <-time.After(30 * time.Second):
 			t.Fatal("timed out waiting for job event")
 		}
 	}
-	t.Assert(actual, c.HasLen, 2)
-	t.Assert(actual[0].Event, c.Equals, host.JobEventCreate)
-	t.Assert(actual[1].Event, c.Equals, host.JobEventError)
-	jobErr := actual[1].Job.Error
+	t.Assert(actual[host.JobEventCreate], c.Not(c.IsNil))
```

Comment:
Should this check that `actual[host.JobEventError]` is not nil, given that is what we use for further assertions below?